libarchive/archive_write_add_filter_zstd.c

```diff
@@ -250,6 +255,18 @@ archive_compressor_zstd_options(struct archive_write_filter *f, const char *key,
 		}
 		data->max_frame_size = max_frame_size;
 		return (ARCHIVE_OK);
+#endif
+#if ZSTD_VERSION_NUMBER >= MINVER_LONG
+	}
+	else if (strcmp(key, "long") == 0) {
+		intmax_t long_distance;
+		if (string_to_number(value, &long_distance) != ARCHIVE_OK) {
+			return (ARCHIVE_WARN);
+		}
+		if (((int)long_distance) < 0)
+		    return (ARCHIVE_WARN);
```

Comment:
You should also check to make sure the value isn't outrageously large.   I presume, for example, that it cannot reasonably be larger than 64?

Reply:
I'm now getting the bounds from the zstd lib, and if that fails I'm using the same logic